scripts/apply_migration.py

```
@@ -0,0 +1,371 @@
+        try:
+            # Check that sources table exists
+            try:
+                self.supabase_client.table("sources").select("source_id").limit(1).execute()
+                logger.info("✓ Sources table exists")
+            except Exception as e:
+                error_msg = str(e).lower()
+                if "relation" in error_msg and "does not exist" in error_msg:
+                    raise MigrationError("Sources table does not exist. Please run initial database setup first.")
+                else:
+                    logger.warning(f"⚠ Could not verify sources table: {e}")
+            
```

Flake8 (7.2.0)

[warning] blank line contains whitespace

(W293)